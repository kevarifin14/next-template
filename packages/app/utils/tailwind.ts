export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};
