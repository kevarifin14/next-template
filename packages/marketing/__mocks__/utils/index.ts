import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
