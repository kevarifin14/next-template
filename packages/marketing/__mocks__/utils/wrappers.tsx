import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
