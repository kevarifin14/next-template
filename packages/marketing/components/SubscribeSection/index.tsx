import { Card } from 'shared';

import Section from 'components/Section';
import SubscribeForm from 'components/SubscribeForm';

type SubscribeSectionProps = {
  title?: string,
  description?: string,
  subscribeCta?: string,
};

export default function SubscribeSection({
  title, description, subscribeCta,
}: SubscribeSectionProps) {
  return (
    <Section size="7xl">
      <Card
        className="grid grid-cols-1 gap-8 lg:grid-cols-2 items-center py-8 px-4 lg:px-16 lg:py-24"
        type="primary"
      >

        <div className="space-y-2">
          <h1 className="text-white text-2xl font-extrabold m-0">
            {title}
          </h1>
          <p className="text-gray-200 m-0">
            {description}
          </p>
        </div>

        <SubscribeForm size="lg" cta={subscribeCta} />

      </Card>
    </Section>
  );
}
