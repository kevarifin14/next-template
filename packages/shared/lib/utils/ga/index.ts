export const pageview = (url) => {
  window.gtag("config", process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS, {
    page_path: url,
  });
};

type EventCategory =
  | "lead"
  | "outbound link"
  | "call to action"
  | "login"
  | "swap";

export const event = (
  category: EventCategory,
  action,
  label = "",
  params = {}
) => {
  if (process.env.NODE_ENV === "production") {
    window.gtag("event", action, {
      event_category: category,
      event_label: label,
      ...params,
    });
  }
};
