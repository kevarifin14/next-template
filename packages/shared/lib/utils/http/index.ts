import QueryString from 'query-string';

export function generateQueryString(params) {
  return QueryString.stringify(params);
}

export function generateEndpoint(endpoint, params) {
  if (Object.keys(params).length === 0) {
    return endpoint;
  }
  return `${endpoint}?${generateQueryString(params)}`;
}

export const get = async (endpoint, params = {}) => {
  const endpointWithParams = generateEndpoint(endpoint, params);
  const response = await fetch(endpointWithParams);

  if (response.status === 200) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};

export const post = async (endpoint, body, headers?) => {
  const response = await fetch(
    endpoint,
    {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'Content-Type': 'application/json',
        ...headers,
      },
    },
  );

  if (response.status >= 200 && response.status < 400) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};
