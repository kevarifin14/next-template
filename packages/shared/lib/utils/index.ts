export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
