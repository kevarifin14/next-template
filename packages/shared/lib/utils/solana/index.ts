/* istanbul ignore file */
import {
  AccountLayout,
  u64,
  AccountInfo as TokenAccountInfo,
  MintInfo,
} from "@solana/spl-token";
import { PublicKey } from "@solana/web3.js";

export const deserializeTokenAccount = (address: PublicKey, data: Buffer) => {
  const {
    mint,
    owner,
    amount,
    delegateOption,
    delegate,
    delegatedAmount,
    state,
    isNative,
    isNativeOption,
    closeAuthority,
    closeAuthorityOption,
  } = AccountLayout.decode(data);

  const accountInfo: TokenAccountInfo = {
    address,
    mint: new PublicKey(mint),
    owner: new PublicKey(owner),
    amount: u64.fromBuffer(amount),
    isInitialized: state !== 0,
    isFrozen: state === 2,

    ...(delegateOption === 0
      ? { delegate: null, delegatedAmount: new u64(0) }
      : {
          delegate: new PublicKey(delegate),
          delegatedAmount: u64.fromBuffer(delegatedAmount),
        }),

    ...(isNativeOption === 1
      ? { rentExemptReserve: u64.fromBuffer(isNative), isNative: true }
      : { rentExemptReserve: null, isNative: false }),

    ...(closeAuthorityOption === 0
      ? { closeAuthority: null }
      : { closeAuthority: new PublicKey(closeAuthority) }),
  };

  return accountInfo;
};

export const diffTokenAccountInfoAmounts = (
  prevTokenAccountInfo: TokenAccountInfo,
  tokenAccountInfo: TokenAccountInfo
) => {
  if (!prevTokenAccountInfo) {
    return tokenAccountInfo.amount;
  }

  return tokenAccountInfo.amount.sub(prevTokenAccountInfo.amount);
};

export const fromLamports = (
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo
) => {
  if (!tokenAccountInfo) {
    return 0;
  }

  const amount =
    typeof tokenAccountInfo === "number"
      ? tokenAccountInfo
      : tokenAccountInfo.amount.toNumber();

  const precision = 10 ** (mintInfo?.decimals || 9);
  return amount / precision;
};

export function formatTokenAmount(
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo,
  prefix = "",
  suffix = "",
  precision = 3
): string {
  return `${[prefix]}${fromLamports(tokenAccountInfo, mintInfo).toFixed(
    precision
  )}${suffix}`;
}

export const formatDollar = (value) => {
  const formatter = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
  });

  return formatter.format(value);
};
