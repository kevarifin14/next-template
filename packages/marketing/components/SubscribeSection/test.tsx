import { render } from '@testing-library/react';

import { notificationsProviderWrapper } from '__mocks__/utils/wrappers';

import SubscribeSection from '.';

describe('SubscribeSection', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<SubscribeSection />, { wrapper: notificationsProviderWrapper });
    expect(baseElement).toMatchSnapshot();
  });
});
