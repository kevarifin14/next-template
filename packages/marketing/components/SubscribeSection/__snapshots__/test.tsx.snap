// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeSection matches snapshot 1`] = `
<body>
  <div>
    <div
      class="fixed max-w-sm w-full right-0 top-0 flex flex-col justify-end items-start space-y-4 pointer-events-none z-30 p-4"
    />
    <div
      class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl"
    >
      <div
        class="px-6 py-8 rounded-3xl shadow-xl dark:shadow-none bg-primary-dark grid grid-cols-1 gap-8 lg:grid-cols-2 items-center py-8 px-4 lg:px-16 lg:py-24"
      >
        <div
          class="space-y-2"
        >
          <h1
            class="text-white text-2xl font-extrabold m-0"
          />
          <p
            class="text-gray-200 m-0"
          />
        </div>
        <form
          class="flex space-x-4"
        >
          <div
            class="w-full"
          >
            <div
              class="relative"
            >
              <input
                class="shadow-sm focus:outline-none w-full border bg-light border-light-400 rounded-md block text-gray-900 placeholder-gray-500 focus:border-primary-600 focus:ring-primary-600 focus:ring-1 dark:bg-dark-light dark:border-transparent dark:text-white px-5 py-3 text-lg"
                id="123456879"
                name="email"
                placeholder="Enter Your Email"
                type="email"
              />
            </div>
          </div>
          <div
            class="flex-shrink-0"
          >
            <button
              class="inline-flex items-center justify-center focus:outline-none font-medium rounded-md text-white bg-gradient-to-tr from-primary-500 to-primary-800    px-5 py-3 text-lg "
              type="submit"
            >
              Get Early Access
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
